File: w02/d02/liza_ramo/Cars/lib/car.rb

class Car

  attr_reader(:horsepower, :fuel)

  def initialize(horsepower, fuel)
    @horsepower = horsepower
    if fuel > 5
      @fuel = 5
    elsif fuel < 0
      @fuel = 0
    else
      @fuel = fuel
    end
  end

  def rev()
    return "VROOM"
  end

  def drive()
    if @fuel > 0
      @fuel -= 1
      return true
    else
      return false
    end
  end

end

File: w10/d04/ranjan_agarwal/cars/car.js
// var carFunctions = {
//   drive: function() {
//     this.fuel--;
//     return 'Vroom';
//   },
//   refuel: function() {
//     this.fuel = 100;
//   }
// };

var Car = function(model, color) {
  this.model = model;
  this.color = color;
  this.fuel = 100;
}

Car.prototype.drive = function() {
  this.fuel--;
  return 'Vroom!';
};

Car.prototype.refuel = function() {
  this.fuel = 100;
};

var dbNine = new Car('Aston Martin DB-9', 'Teal');
